File: CS/NavBarSample/CustomNavigationPaneView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.Xpf.NavBar;

namespace NavBarSample {
    public class CustomNavigationPaneView : NavigationPaneView {
        public double OpenedWidth { get; set; }
        public double PreviousWidth { get; set; }
        protected override void UpdatePresenterTemplateByEnabledState() {
            if (OpenedWidth != 0) {
                ExpandedWidth = OpenedWidth;
            }
            base.UpdatePresenterTemplateByEnabledState();
        }       
    }
}

File: VB/NavBarSample/CustomNavigationPaneView.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.Xpf.NavBar

Namespace NavBarSample
    Public Class CustomNavigationPaneView
        Inherits NavigationPaneView

        Public Property OpenedWidth() As Double
        Public Property PreviousWidth() As Double
        Protected Overrides Sub UpdatePresenterTemplateByEnabledState()
            If OpenedWidth <> 0 Then
                ExpandedWidth = OpenedWidth
            End If
            MyBase.UpdatePresenterTemplateByEnabledState()
        End Sub
    End Class
End Namespace
